common/lib/xmodule/xmodule/js/src/capa/display.coffee

```diff
@@ -868,8 +872,23 @@ class @Problem
       next_index = parseInt(hint_index) + 1
     $.postWithPrefix "#{@url}/hint_button", hint_index: next_index, input_id: @id, (response) =>
       if response.success
-        @render(response.html, @focus_on_hint_notification)
-        hint_container = @.$('.problem-hint')
+        hint_msg_container = @.$('.problem-hint .notification-message')
         hint_container.attr('hint_index', response.hint_index)
+        hint_msg_container.html(response.msg)
+        # Update any Mathjax entries
+        MathJax.Hub.Queue [
+          'Typeset'
+          MathJax.Hub
+          hint_container[0]
+        ]
+        @hintNotification.show()
+        # Enable/Disable the next hint button
+        if response.more_hints
```

Comment:
Let's update the button status before the show, to avoid any possible flashing.

And both buttons have a common class, so you should be able to update them in one statement, using just @hintButton (which is already using this fact to attach the click listener to both).